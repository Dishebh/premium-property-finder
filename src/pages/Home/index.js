import * as React from 'react'
import { Link } from '@reach/router'

function Home() {
  return (
    <>
      <h1>Home</h1>
      <Link to="/details/123">Details</Link>
    </>
  )
}

export default Home
