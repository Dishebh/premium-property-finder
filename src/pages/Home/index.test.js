import * as React from 'react'
import { render, cleanup } from 'react-testing-library'

import 'jest-dom/extend-expect'
import renderer from 'react-test-renderer'

import Home from '.'

beforeAll(() => {
  global.alert = jest.fn()
})

afterEach(cleanup)

test('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('should render basic elements', () => {
  const { getByText } = render(<Home />)
  expect(getByText(/Premium Property Finder/i)).toBeTruthy()
  expect(getByText(/Bringing premium property right to your finger tips/i)).toBeTruthy()
})
