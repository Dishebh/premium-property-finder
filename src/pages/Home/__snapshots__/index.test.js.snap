// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <main
    className="mb-3"
    role="main"
  >
    <div
      className="hero hero mb-3 hero-lg"
    >
      <div
        className="hero-body text-center text-light"
      >
        <h1>
          Premium Property Finder
        </h1>
        <p
          className="mb-0"
        >
          Bringing premium property right to your finger tips
        </p>
      </div>
    </div>
    <div
      className="container"
    >
      <aside
        className="mt-2"
      >
        <div
          className="container mb-1 container"
        >
          <h2>
            0 private properties for sale
          </h2>
          <form
            className="form-horizontal form"
            noValidate={true}
            onChange={[Function]}
          >
            <p
              className="mb-1"
            >
              Refine your results
              <button
                className="ml-1 btn btn-sm"
                onClick={[Function]}
                type="button"
              >
                Clear
              </button>
            </p>
            <div
              className="columns text-center"
            >
              <div
                className="column col-4 col-xs-12"
              >
                <div
                  className="form-group"
                >
                  <div
                    className="col-3 col-sm-12"
                  >
                    <label
                      className="form-label"
                      htmlFor="price-from"
                    >
                      Price from
                    </label>
                  </div>
                  <div
                    className="col-9 col-sm-12"
                  >
                    <input
                      className="form-input"
                      id="price-from"
                      max="10000000"
                      min="0"
                      onChange={[Function]}
                      placeholder="Â£1,000,000"
                      type="number"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                className="column col-4 col-xs-12"
              >
                <div
                  className="form-group"
                >
                  <div
                    className="col-3 col-sm-12"
                  >
                    <label
                      className="form-label"
                      htmlFor="postcode"
                    >
                      Postcode
                    </label>
                  </div>
                  <div
                    className="col-9 col-sm-12"
                  >
                    <select
                      className="form-select"
                      id="postcode"
                      onChange={[Function]}
                      value=""
                    >
                      <option
                        value=""
                      >
                        Choose...
                      </option>
                    </select>
                  </div>
                </div>
              </div>
              <div
                className="column col-4 col-xs-12"
              >
                <div
                  className="form-group"
                >
                  <div
                    className="col-3 col-sm-12"
                  >
                    <label
                      className="form-label"
                      htmlFor="sortorder"
                    >
                      Sort Order
                    </label>
                  </div>
                  <div
                    className="col-9 col-sm-12"
                  >
                    <select
                      className="form-select"
                      id="sortorder"
                      onChange={[Function]}
                      value=""
                    >
                      <option
                        value=""
                      >
                        Choose...
                      </option>
                      <option
                        value="highestfirst"
                      >
                        Highest First
                      </option>
                      <option
                        value="lowestfirst"
                      >
                        Lowest First
                      </option>
                    </select>
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
      </aside>
      <div
        className="columns"
      />
    </div>
  </main>,
  <footer
    className="text-center mb-5"
  >
    Developed By
     
    <a
      href="https://twitter.com/jpreecedev"
      rel="noopener noreferrer"
      target="_blank"
    >
      Jon Preece
    </a>
    , 2019
  </footer>,
]
`;
